delphi/umap_narrative/502_calculate_priorities.py

```diff
@@ -0,0 +1,320 @@
+        """
+        Calculate importance metric (matches Clojure implementation).
+        
+        Args:
+            A: Number of agree votes
+            P: Number of pass votes  
+            S: Total number of votes
+            E: Extremity value
+            
+        Returns:
+            Importance metric value
+        """
+        # Laplace smoothing
+        p = (P + 1) / (S + 2)  # Pass rate
+        a = (A + 1) / (S + 2)  # Agree rate
+        
+        # Importance calculation: (1 - p) * (E + 1) * a
+        return (1 - p) * (E + 1) * a
+
+    def _priority_metric(self, is_meta: bool, A: int, P: int, S: int, E: float) -> float:
+        """
+        Calculate priority metric (matches Clojure implementation).
+        
+        Args:
+            is_meta: Whether the comment is a meta comment
+            A: Number of agree votes
+            P: Number of pass votes
+            S: Total number of votes
+            E: Extremity value
+            
+        Returns:
+            Priority metric value
+        """
+        META_PRIORITY = 7.0
+        
+        if is_meta:
+            return META_PRIORITY ** 2  # 49
+        else:
+            # Regular priority calculation
+            importance = self._importance_metric(A, P, S, E)
+            # Scale by a factor which lets new comments bubble up
+            scaling_factor = 1.0 + (8.0 * (2.0 ** (-S / 5.0)))
+            return (importance * scaling_factor) ** 2
+
+    def get_comment_extremity(self, comment_id: str) -> float:
+        """
+        Get extremity value for a comment from DynamoDB.
+        
+        Args:
+            comment_id: The comment ID
+            
+        Returns:
+            Extremity value (0.0 to 1.0) or 0.0 if not found
+        """
+        try:
+            response = self.comment_extremity_table.get_item(
+                Key={
+                    'conversation_id': str(self.conversation_id),
+                    'comment_id': str(comment_id)
+                }
+            )
+            
+            if 'Item' in response:
+                extremity_value = response['Item'].get('extremity_value', 0.0)
+                return float(extremity_value)
+            else:
+                logger.debug(f"No extremity data found for comment {comment_id}")
+                return 0.0
+                
+        except Exception as e:
+            logger.warning(f"Error retrieving extremity for comment {comment_id}: {e}")
+            return 0.0
+
+    def get_comment_routing_data(self) -> List[Dict[str, Any]]:
+        """
+        Get all comment routing data for the conversation.
+        
+        Returns:
+            List of comment routing items
+        """
+        try:
+            # Scan for all items with this conversation ID
+            response = self.comment_routing_table.scan(
+                FilterExpression='contains(zid_tick, :zid)',
+                ExpressionAttributeValues={':zid': str(self.conversation_id)}
+            )
```

Comment:
we'll have to change this to avoid scanning the entire table. A GSI of ZID will need to be added to this table, which will then enable finding items by ZID directly